static/app/utils/makeSafeRefluxStore.ts

```diff
@@ -0,0 +1,50 @@
+export interface SafeRefluxStore extends Reflux.Store {
+  teardown(): void;
+  unsubscribeListeners: Reflux.Subscription[];
+}
+
+export function makeSafeRefluxStore<T extends Reflux.Store>(
+  store: T
+): SafeRefluxStore & T {
+  const originalListenTo = store.listenTo.bind(store);
+  const safeStore = store as SafeRefluxStore & T;
+
+  // Warning: We are going to be overriding instance methods here!
```

Comment:
```suggestion
  // Warning: We are going to be overriding instance methods here!
```

nit: I typically write these as `XXX:` https://stackoverflow.com/a/7284583/790169